From 5dcce5673437571660069e30c15be6ecd21e586e Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:52 -0400
Subject: [scsi] qla2xxx: Fix beacon blink logic for ISP26xx/83xx

Message-id: <1408391817-6389-65-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87946
O-Subject: [PATCH RHEL 7.1 e-stor 64/69] qla2xxx: Fix beacon blink logic for ISP26xx/83xx.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Himanshu Madani <himanshu.madani@qlogic.com>

scsi commit id 90b604f26131eb3a76a1588538ec96c7148e6507

Signed-off-by: Himanshu Madani <himanshu.madani@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 68299d0..d1de306 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -4974,8 +4974,10 @@ intr_on_check:
    ha->isp_ops->enable_intrs(ha);
 
   if (test_and_clear_bit(BEACON_BLINK_NEEDED,
-     &base_vha->dpc_flags))
-   ha->isp_ops->beacon_blink(base_vha);
+     &base_vha->dpc_flags)) {
+   if (ha->beacon_blink_led == 1)
+    ha->isp_ops->beacon_blink(base_vha);
+  }
 
   if (!IS_QLAFX00(ha))
    qla2x00_do_dpc_all_vps(base_vha);
diff --git a/drivers/scsi/qla2xxx/qla_sup.c b/drivers/scsi/qla2xxx/qla_sup.c
index c2362da..7897005 100644
--- a/drivers/scsi/qla2xxx/qla_sup.c
+++ b/drivers/scsi/qla2xxx/qla_sup.c
@@ -1702,11 +1702,8 @@ qla83xx_beacon_blink(struct scsi_qla_host *vha)
  if (IS_QLA2031(ha)) {
   led_select_value = qla83xx_select_led_port(ha);
 
-  qla83xx_wr_reg(vha, led_select_value, 0x40002000);
-  qla83xx_wr_reg(vha, led_select_value + 4, 0x40002000);
-  msleep(1000);
-  qla83xx_wr_reg(vha, led_select_value, 0x40004000);
-  qla83xx_wr_reg(vha, led_select_value + 4, 0x40004000);
+  qla83xx_wr_reg(vha, led_select_value, 0x40000230);
+  qla83xx_wr_reg(vha, led_select_value + 4, 0x40000230);
  } else if (IS_QLA8031(ha)) {
   led_select_value = qla83xx_select_led_port(ha);
 
-- 
1.7.1